test/jasmine/tests/click_test.js

```diff
@@ -817,6 +928,588 @@ describe('Test click interactions:', function() {
+                xvals0 = futureData.xvals[0],
+                yaxes0 = futureData.yaxes[0],
+                yvals0 = futureData.yvals[0];
+
+            expect(Object.keys(pt)).toEqual([
+                'data', 'fullData', 'curveNumber', 'pointNumber', 'x', 'y',
+                'xaxis', 'yaxis'
+            ]);
+
+            expect(pt.curveNumber).toEqual(0, 'points[0].curveNumber');
+            expect(typeof pt.data).toEqual(typeof {}, 'points[0].data');
+            expect(typeof pt.fullData).toEqual(typeof {}, 'points[0].fullData');
+            expect(pt.pointNumber).toEqual(0, 'points[0].pointNumber');
+            expect(pt.x).toEqual(undefined, 'points[0].x');
+            expect(pt.y).toEqual(undefined, 'points[0].y');
+            expect(typeof pt.xaxis).toEqual(typeof {}, 'points[0].xaxis');
+            expect(typeof pt.yaxis).toEqual(typeof {}, 'points[0].yaxis');
+
+            expect(xaxes0).toEqual(pt.xaxis, 'xaxes[0]');
+            expect(xvals0).toEqual(-0.0016654247744483342, 'xaxes[0]');
+            expect(yaxes0).toEqual(pt.yaxis, 'yaxes[0]');
+            expect(yvals0).toEqual(0.5013, 'xaxes[0]');
+
+            expect(evt.clientX).toEqual(pointPos[0], 'event.clientX');
+            expect(evt.clientY).toEqual(pointPos[1], 'event.clientY');
+        });
+    });
+
+    describe('unhover events', function() {
+        var futureData;
+
+        beforeEach(function(done) {
+            Plotly.plot(gd, mockCopy.data, mockCopy.layout).then(done);
+
+            gd.on('plotly_unhover', function(data) {
+                futureData = data;
+            });
+        });
+
+        it('should contain the correct fields', function() {
+            mouseEvent('mousemove', blankPos[0], blankPos[1]);
+            mouseEvent('mousemove', pointPos[0], pointPos[1]);
+            mouseEvent('mouseout', pointPos[0], pointPos[1]);
+
+            var pt = futureData.points[0],
+                evt = futureData.event;
+
+            expect(Object.keys(pt)).toEqual([
+                'data', 'fullData', 'curveNumber', 'pointNumber', 'x', 'y',
+                'xaxis', 'yaxis'
+            ]);
+
+            expect(pt.curveNumber).toEqual(0, 'points[0].curveNumber');
+            expect(typeof pt.data).toEqual(typeof {}, 'points[0].data');
+            expect(typeof pt.fullData).toEqual(typeof {}, 'points[0].fullData');
+            expect(pt.pointNumber).toEqual(0, 'points[0].pointNumber');
+            expect(pt.x).toEqual(undefined, 'points[0].x');
+            expect(pt.y).toEqual(undefined, 'points[0].y');
+            expect(typeof pt.xaxis).toEqual(typeof {}, 'points[0].xaxis');
+            expect(typeof pt.yaxis).toEqual(typeof {}, 'points[0].yaxis');
+
+            expect(evt.clientX).toEqual(pointPos[0], 'event.clientX');
+            expect(evt.clientY).toEqual(pointPos[1], 'event.clientY');
+        });
+    });
+});
+
+
+describe('Test click interactions on a geo plot:', function() {
```

Comment:
Let's move these to https://github.com/plotly/plotly.js/blob/master/test/jasmine/tests/geo_test.js